src/libraries/System.IO.Compression/tests/ZipArchive/zip_UpdateTests.Comments.cs

```diff
@@ -11,179 +11,86 @@ namespace System.IO.Compression.Tests
     public partial class zip_UpdateTests : ZipFileTestBase
     {
         [Theory]
-        [MemberData(nameof(Encoding_GetData))]
-        public static void Update_ZipArchive_Comment(
-            Encoding constructorEncoding,
-            string originalComment, string expectedOriginalComment,
-            string updatedComment, string expectedUpdatedComment)
-        {
-            // The archive comment max size in bytes is determined by the internal
-            // ZipEndOfCentralDirectoryBlock.ZipFileCommentMaxLength const field.
-            // The current value of that field is set to ushort.MaxValue.
-
-            using var stream = new MemoryStream();
-            var testStream = new WrappedStream(stream, true, true, true, null);
-
-            // Create
-            using (var zip = new ZipArchive(testStream, ZipArchiveMode.Create, leaveOpen: true, constructorEncoding))
-            {
-                zip.Comment = originalComment;
-                Assert.Equal(expectedOriginalComment.Length, zip.Comment.Length);
-                Assert.Equal(expectedOriginalComment, zip.Comment);
-            }
-            // Read (verify creation)
-            using (var zip = new ZipArchive(testStream, ZipArchiveMode.Read, leaveOpen: true, constructorEncoding))
-            {
-                Assert.Equal(expectedOriginalComment.Length, zip.Comment.Length);
-                Assert.Equal(expectedOriginalComment, zip.Comment);
-            }
-            // Update
-            using (var zip = new ZipArchive(testStream, ZipArchiveMode.Update, leaveOpen: true, constructorEncoding))
-            {
-                zip.Comment = updatedComment;
-                Assert.Equal(expectedUpdatedComment.Length, zip.Comment.Length);
-                Assert.Equal(expectedUpdatedComment, zip.Comment);
-            }
-            // Read (verify update) and close stream
-            using (var zip = new ZipArchive(testStream, ZipArchiveMode.Read, leaveOpen: false, constructorEncoding))
-            {
-                Assert.Equal(expectedUpdatedComment.Length, zip.Comment.Length);
-                Assert.Equal(expectedUpdatedComment, zip.Comment);
-            }
-        }
+        [MemberData(nameof(Utf8Comment_Data))]
+        public static void Update_Comment_AsciiEntryName_NullEncoding(string originalComment, string expectedComment) =>
+            Update_Comment_EntryName_Encoding_Internal("file.txt",
+                originalComment, expectedComment, null,
+                new string('a', ushort.MaxValue - 1) + $"{CopyrightChar}", new string('a', ushort.MaxValue - 1));
```

Comment:
This is testing `Utf8Comment_Data.Count()` times the string `new string('a', ushort.MaxValue - 1) + $"{CopyrightChar}"`, you should create another Utf8Comment_Data that includes this as a parameter and remove the redundancy.